### PR TITLE
[IMP] send_quotations/pdf_quote_builder

### PR DESCRIPTION
**Pull request by Demesmaeker**
Add documentation about the new features and possibilities of the PDF quote builder. Delete obsolote information.

### CODE REVIEW

**Review by StraubCreative — APPROVED** [review]
LGTM off a quick look, @pabr-odoo can you confirm the functional details please so we can merge quickly?

File: content/applications/sales/sales/send_quotations/pdf_quote_builder.rst [hunk 1]
```diff
@@ -30,28 +30,45 @@ To do that, navigate to :menuselection:`Sales app --> Configuration --> Settings
...
 
-The files uploaded here will be the default PDF used for all quotes.
+Under this, in the :guilabel:`Quotation Templates` section, you can restrict this PDF to some  quotation
+templates only.
 
 .. note::
-   Values set in the PDF Quote Builder settings are company-specific.
+   Alternatively, you can also navigate to
+   :menuselection:`Sales app --> Configuration --> Quotation Templates`, select a template and
+   directly :guilabel:`Add` or :guilabel:`Upload` a PDF to it.
```
Comment (pabr-odoo) [inline content/applications/sales/sales/send_quotations/pdf_quote_builder.rst:68]:
```suggestion
   directly :guilabel:`Add` or :guilabel:`Upload` a PDF to it in the :guilabel:`Quote Builder` tab.
```

File: content/applications/sales/sales/send_quotations/pdf_quote_builder.rst [hunk 2]
```diff
@@ -93,6 +109,13 @@ Product-specific dynamic text values are as follows:
 - :guilabel:`tax_excl_price`: Tax Excluded Price
 - :guilabel:`tax_incl_price`: Tax Included Price
 
+After uploading a PDF, you can then :guilabel:`Configure dynamic fields`. This will allow you to map
+any field name found in your PDF to the field you want to show by writing down any existing path.
+Headers and footers starts from the current :guilabel:`sale_order` model, whereas product document
```
Comment (pabr-odoo) [inline content/applications/sales/sales/send_quotations/pdf_quote_builder.rst:114]: I'm not quite sure what this sentence is referring to. I cannot find sale_order or sale_order_line in the Form Fields page for a header.

File: content/applications/sales/sales/send_quotations/pdf_quote_builder.rst [hunk 3]
```diff
@@ -106,11 +129,24 @@ Product-specific dynamic text values are as follows:
...
+   configuring the :guilabel:`path` of the :guilabel:`Form Field Name` to
+   :guilabel:`partner_invoice_id.country_id.name` fills the form with the invoice partner country's
+   name when the PDF is built.
 
-Once the upload(s) are complete, click :guilabel:`Save`.
+.. example::
+   When uploading a product PDF containing the form field :guilabel:`invoice_partner_country`,
+   configuring the :guilabel:`path` of the :guilabel:`Form Field Name` to
+   :guilabel:`order_id.partner_invoice_id.country_id.name` fills the form with the invoice partner
+   country's name when the PDF is built.
```
Comment (pabr-odoo) [inline content/applications/sales/sales/send_quotations/pdf_quote_builder.rst:144]: I believe these two examples are duplicates.

File: content/applications/sales/sales/send_quotations/pdf_quote_builder.rst [hunk 4]
```diff
@@ -188,6 +224,11 @@ quote`.
        :align: center
        :alt: Sample of an uploaded pdf with the inside quote option chosen in Odoo Sales.
 
+Beside the :guilabel:`File Content` field, you have the possibility to
+:guilabel:`Configure dynamic fields`. When doing so, remember that the starting model is the
+:guilabel:`sale_order_line`, unlike for headers and footers that start from the
+:guilabel:`sale_order`.
```
Comment (pabr-odoo) [inline content/applications/sales/sales/send_quotations/pdf_quote_builder.rst:230]: Similar to the previous comment, I can't find sale_order_line or sale_order in the UI.

File: content/applications/sales/sales/send_quotations/pdf_quote_builder.rst [hunk 5]
```diff
@@ -205,6 +246,13 @@ page` on the front-end (in the online store).
 PDF quote
 =========
 
+On a :guilabel:`Sale Order`, in the tab :guilabel:`Quote Builder`, select any document you want to
```
Comment (pabr-odoo) [inline content/applications/sales/sales/send_quotations/pdf_quote_builder.rst:249]:
```suggestion
On a :guilabel:`Sale Order`, in the :guilabel:`Quote Builder` tab, select any document you want to
```

**Review by pabr-odoo — COMMENTED** [review]
Hi @Demesmaeker I finished reviewing the functional details of this document and it looks great overall! I left a few suggestions for you to review but otherwise everything functions as it should. Thanks!

File: content/applications/sales/sales/send_quotations/pdf_quote_builder.rst (hunk 3 above)
Comment (Demesmaeker) [reply content/applications/sales/sales/send_quotations/pdf_quote_builder.rst:144]: The difference between these two is that one is a header document, whereas the other is a product document. I voluntarily used the same final information to point the different path, as the path is different for product or header/footer document.
I added some formatting to improve the readability/difference between the two.

File: content/applications/sales/sales/send_quotations/pdf_quote_builder.rst [hunk 6]
```diff
@@ -106,11 +129,24 @@ Product-specific dynamic text values are as follows:
...
+.. example::
+   When uploading a **product** PDF containing the form field :guilabel:`invoice_partner_country`,
+   configuring the :guilabel:`path` of the :guilabel:`Form Field Name` to
+   :guilabel:`order_id.partner_invoice_id.country_id.name` fills the form with the invoice partner
+   country's name when the PDF is built.
+
+.. example::
+   When uploading any PDF containing the form field :guilabel:`custom_note`, leaving the
+   :guilabel:`path` empty allows the seller to write down any note where that form field is in that
+   document and shown when the PDF is built.
```
Comment (pabr-odoo) [inline content/applications/sales/sales/send_quotations/pdf_quote_builder.rst:149]: A totally optional suggestion here is using a single example attribution, with the content of each example block as bullet points.

File: content/applications/sales/sales/send_quotations/pdf_quote_builder.rst [hunk 7]
```diff
@@ -205,6 +246,13 @@ page` on the front-end (in the online store).
 PDF quote
 =========
 
+On a :guilabel:`Sale Order`, in the :guilabel:`Quote Builder` tab, select any document you want to
```
Comment (pabr-odoo) [inline content/applications/sales/sales/send_quotations/pdf_quote_builder.rst:249]:
```suggestion
On a :guilabel:`Sales Order`, in the :guilabel:`Quote Builder` tab, select any document you want to
```

**Review by pabr-odoo — APPROVED** [review]
Hi @Demesmaeker just gave this another look, and it looks great! I left one formatting suggestion and one UI label correction. After you take a look, feel free to merge this doc. Thanks for the great work!

**Review by Feyensv — APPROVED** [review]
@robodoo delegate+

**Comment by Demesmaeker** [comment]
@robodoo r-
missing delete of unused files

**Comment by Demesmaeker** [comment]
@robodoo r+

**Comment by Demesmaeker** [comment]
@robodoo r+
(deleted a bit too much, the settings is still there, the image isn't adequate any more, thus was deleted though)

**Pull request closed without merging**